src/compiler/checker.ts

```diff
@@ -9187,7 +9187,8 @@ namespace ts {
                     }
                 }
             }
-            return createArrayType(elementTypes.length ? getUnionType(elementTypes) : emptyArrayElementType);
+            const elementType = getUnionType(elementTypes);
+            return createArrayType(elementType === neverType ? strictNullChecks ? wideningNeverType : undefinedType : elementType);

```

Comment:
So if you have an array of `[a, b, c]`, and `a`, `b`, and `c` each have type `never`, does the type of the array literal get widened to `any[]`?